src/coreComponents/fieldSpecification/SourceFluxStatistics.cpp

```diff
@@ -0,0 +1,364 @@
+
+  registerWrapper( viewKeyStruct::fluxNamesString().data(), &m_fluxNames ).
+    setRTTypeName( rtTypes::CustomTypes::groupNameRefArray ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setSizedFromParent( 0 ).
+    setDescription( GEOS_FMT( "Name(s) array of the {0}(s) for which we want the statistics. "
+                              "Use \"all\" to target all {0}.",
+                              SourceFluxBoundaryCondition::catalogName() ) );
+}
+
+void SourceFluxStatsAggregator::postProcessInput()
+{
+  Base::postProcessInput();
+
+  FieldSpecificationManager & fsManager = FieldSpecificationManager::getInstance();
+  if( m_fluxNames.size() == 1 && m_fluxNames[0] == "all" )
+  {
+    m_fluxNames.clear();
+    fsManager.forSubGroups< SourceFluxBoundaryCondition >( [&]( SourceFluxBoundaryCondition & sourceFlux )
+    {
+      m_fluxNames.emplace_back( string( sourceFlux.getName() ) );
+    } );
+    GEOS_WARNING_IF( m_fluxNames.empty(),
+                     GEOS_FMT( "{}: No {} was found in {}.",
+                               getDataContext(), SourceFluxBoundaryCondition::catalogName(),
+                               fsManager.getDataContext() ) );
+  }
+  else
+  {
+    for( string const & fluxName : m_fluxNames )
+    {
+      GEOS_ERROR_IF( !fsManager.hasGroup< SourceFluxBoundaryCondition >( fluxName ),
+                     GEOS_FMT( "{}: No {} named {} was found in {}.",
+                               getDataContext(), SourceFluxBoundaryCondition::catalogName(),
+                               fluxName, fsManager.getDataContext() ) );
+    }
+  }
+}
+
+Wrapper< SourceFluxStatsAggregator::WrappedStats > &
+SourceFluxStatsAggregator::registerWrappedStats( Group & group,
+                                                 string_view fluxName,
+                                                 string_view elementSetName )
+{
+  string const wrapperName = getStatWrapperName( fluxName );
+  Wrapper< WrappedStats > & statsWrapper = group.registerWrapper< WrappedStats >( wrapperName );
+  statsWrapper.setRestartFlags( RestartFlags::NO_WRITE );
+  statsWrapper.reference().setTarget( getName(), fluxName );
+
+  writeStatsToCSV( elementSetName, statsWrapper.reference(), true );
+
+  return statsWrapper;
+}
+void SourceFluxStatsAggregator::registerDataOnMesh( Group & meshBodies )
+{
+  if( m_solver == nullptr )
+  {
+    return;
+  }
+
+  m_solver->forDiscretizationOnMeshTargets( meshBodies, [&] ( string const &,
+                                                              MeshLevel & mesh,
+                                                              arrayView1d< string const > const & )
+  {
+    registerWrappedStats( mesh, viewKeyStruct::fluxSetWrapperString(), viewKeyStruct::allRegionWrapperString() );
+
+    for( string const & fluxName : m_fluxNames )
+    {
+      registerWrappedStats( mesh, fluxName, viewKeyStruct::allRegionWrapperString() );
+
+      mesh.getElemManager().forElementRegions( [&]( ElementRegionBase & region )
```

Comment:
If the `logLevel` is low enough, then we don't need to create these wrappers.

Reply:
I use them to collect stats from the `SinglePhaseBase::applySourceFluxBC()` methods that have no knowledge of the regions. Can I keep them to collect the data or should I proceed differently?